### DISCUSSION TITLE
IMB Barcode Reader

### DISCUSSION THREAD

**Discussion by jiteshdhamaniya** [micjahn/ZXing.Net#500] [General]
I could not find documentation to read IMB Barcode. could someone Please share sample code to read IMB Barcode from an image for latest version which is 0.16.9 ?
I am beginner level in Asp.net and using Mac. 

Any help is really appreciated. 

@micjahn 

Thanks

**Comment by micjahn** [comment]
You can start with the following snippet:

```
            BarcodeReader reader = new BarcodeReader()
            {
                Options = new DecodingOptions()
                {
                    PossibleFormats = new List<BarcodeFormat>()
                    {
                        BarcodeFormat.IMB
                    },
                },
            };
            reader.Decode(barcodeBitmapInstance);
```

**Reply by hugueds** [reply]
"Using the generic type 'BarcodeReader<T>' requires 1 type argument"
